core/rest-client/src/http_client.rs

```diff
@@ -432,4 +433,66 @@ mod tests {
 			Error::DeserializeParseError(err, String::from_utf8_lossy(encoded_body).to_string())
 		})
 	}
+
+	#[test]
+	fn get_from_site_with_self_signed_certificate_fails() {
+		let base_url = Url::parse("https://self-signed.badssl.com").unwrap();
+		let result = get_for_test_certificates(base_url);
+		assert_matches!(result, Err(Error::HttpReqError(_)));
```

Comment:
How do we know here that the test fails due to an insecure certificate and not because the site is down? (same for all the tests below)

Reply:
For this test I rework the error check. 
For the let's encrypt test, I will do later in the task https://app.zenhub.com/workspaces/integritee-60ab46e9e8afdd0011cea78d/issues/integritee-network/worker/583

Reply:
Cool, thanks a lot! 
Some follow up questions though:
1. Could you add a description to #583 what's the problem? Just from the title alone, I don't get why you can't add a an assert like you did in this test.
2. You now have a `println` in this test. Why? And is it really necessary to format the error into a String for asserting the correct error has been returned?

Reply:
1: I don't have time to do it now. I have to go deeper in the let's encrypt certificates and check the test server.